iroh-net/src/test_utils.rs

```diff
@@ -1,38 +1,98 @@
 //! Internal utilities to support testing.
 
+use tokio::runtime::RuntimeFlavor;
 use tracing::level_filters::LevelFilter;
+use tracing_subscriber::fmt::format::FmtSpan;
 use tracing_subscriber::layer::{Layer, SubscriberExt};
 use tracing_subscriber::util::SubscriberInitExt;
 use tracing_subscriber::EnvFilter;
 
-/// Configures logging for the current test.
+/// Configures logging for the current test, **single-threaded runtime only**.
+///
+/// This setup can be used for any sync test or async test using a single-threaded tokio
+/// runtime (the default).  For multi-threaded runtimes use [`with_logging`].
 ///
 /// This configures logging that will interact well with tests: logs will be captured by the
 /// test framework and only printed on failure.
 ///
 /// The logging is unfiltered, it logs all crates and modules on TRACE level.  If that's too
 /// much consider if your test is too large (or write a version that allows filtering...).
+///
+/// # Example
+///
+/// ```no_run
+/// #[tokio::test]
+/// async fn test_something() {
+///     let _guard = crate::test_utils::setup_logging();
+///     assert!(true);
+/// }
 #[must_use = "The tracing guard must only be dropped at the end of the test"]
 #[allow(dead_code)]
 pub(crate) fn setup_logging() -> tracing::subscriber::DefaultGuard {
+    if let Ok(handle) = tokio::runtime::Handle::try_current() {
+        match handle.runtime_flavor() {
+            RuntimeFlavor::CurrentThread => (),
+            RuntimeFlavor::MultiThread => {
+                panic!(
+                    "setup_logging() does not work in a multi-threaded tokio runtime, \
+                        use with_logging() instead"
```

Comment:
`with_logging` is commented out?

Reply:
yes, it doesn't want to work :(  needs more debugging

Reply:
then maybe change the panic message, it's a bit confusing atm 😅

Reply:
done